WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```diff
@@ -0,0 +1,762 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using DynamicData;
+using ReactiveUI;
+using WoWsShipBuilder.Core.DataContainers;
+using WoWsShipBuilder.Core.DataProvider;
+using WoWsShipBuilder.Core.Localization;
+using WoWsShipBuilder.Core.Services;
+using WoWsShipBuilder.Core.Settings;
+using WoWsShipBuilder.DataStructures.Ship;
+using WoWsShipBuilder.ViewModels.Base;
+using WoWsShipBuilder.ViewModels.Helper;
+
+namespace WoWsShipBuilder.ViewModels.Other;
+
+public class ShipComparisonViewModel : ViewModelBase
+{
+    public const string DataNotAvailable = "N/A";
+
+    public static readonly string DefaultBuildName = "---";
+
+    private readonly IAppDataService appDataService;
+
+    private readonly ILocalizer localizer;
+
+    private readonly AppSettings appSettings;
+
+    private readonly IEnumerable<Ship> fullShipList = AppData.ShipDictionary!.Values;
+
+    private List<ShipComparisonDataWrapper> filteredShipList = new();
```

Comment:
Instead of using multiple List<ShipComparisonDataWrapper> objects where data may be accidentally duplicated, use one source of truth like a Dictionary<Guid, ShipComparisonDataWrapper> and then use only List<Guid> to store IDs for the wrappers. That way, complex objects like the wrappers are stored in one place only and easier to access.